src/Plant.py

```diff
@@ -20,7 +22,7 @@ def setLast_watered(self,new_watering):
         self.__last_watered = new_watering
 
     #When obj is initialized, this sets a name, period between watering, what time the plant is watered, and when the plant was last watered
-    def __init__(self,plant_name, plant_watering_interval, plant_watering_time, plant_last_watered):
+    def __init__(self,plant_name = "Undefined", plant_watering_interval = datetime.timedelta(1), plant_watering_time = datetime.datetime.now().time(), plant_last_watered = datetime.datetime.now()):
```

Comment:
I recommend leaving the default off for `plant_name`. That would make it a required parameter, forcing the user to name the plant and avoiding accidental name collisions (we can check for duplicate names later, but we still don't want to get into the habit of having defaults that would result in collisions).

Reply:
Okay will do! Should I leave off the other parameters as well?

Reply:
This is a good opportunity to really think about the role of the Plant object here!

What of these parameters do we actually need when we first create a Plant object? What are the _essential_ items, do you think?

Optional parameters, particularly in constructors like this, generally contain what are known as "sane defaults." In the context of a plant, that might be something like "water every 2 days."

Of these parameters, what ones (if any) are most likely to be `None` _most_ of the time?

Additionally, what of these parameters are _okay to be_ `None` for an indefinite period of time?

Also, which ones describe _attributes_ of the Plant? If any _aren't_ attributes of the Plant, itself, what are they?